### DISCUSSION TITLE
Covariance matrix is displayed incorrectly in the report file

### DISCUSSION THREAD

**Discussion by Peku995** [cdslaborg/paramonte#36] [Q&A] [answered]
Hello everyone,

I have noticed that my input covariance/correlation matrix are displayed incorrectly in the report file. The first row corresponds to the matrix I entered, but all the others are from the default unit matrix. 
I don't know whether I'm doing something wrong or whether it's possibly a display problem in the report file. 
I am using Matlab.

Best regards

**Comment by shahmoradi** [comment]
@Peku995 
Thanks for raising this question.
How many dimensions is your covariance?
My first suggestion is to,
1. specify only covariance **in full** (with all elements), or
2. specify only the correlation matrix and standard deviation **in full** (without setting covariance as input).

The covariance computation strategy is involved in the library, meaning it depends on several other quantities you can set and is done element-wise.
Note that specifying covariance as input has precedence over specifying the correlation matrix.
If you specify both simultaneously, only the covariance matrix will be used.
The ability to set the correlation matrix and standard deviations (besides covariance) is there to simplify input data merely, often available as a correlation matrix and standard deviation vector. 

The new, enhanced MATLAB version of the library is being tested for release sometime next week.  
We plan on releasing the C/C++/Fortran versions likely by tonight.  
If you are *watching* the repository, you will likely receive notifications when the new version is released in the main branch.

**Reply by Peku995** [reply]
Thanks for the answer.
I think I did what you said because I specified the covariance matrix as the following 6x6 matrix (while I tried so sample 6 variables) 
![image](https://github.com/cdslaborg/paramonte/assets/137407148/b626392f-50f3-4490-842b-e38f332f1775)
What I got back in the report file was 
![image](https://github.com/cdslaborg/paramonte/assets/137407148/bb47af71-a942-4f4a-9452-2c30f27e3687)
Therefore, maybe Paramonte does not visualize the input covariance matrix correctly.

The new maltab version is great news, looking forward!

Best regards

**Comment by shahmoradi** [comment, marked as answer]
Hi @Peku995, 
Try the second option and see if that fixes the issue by specifying the correlation matrix and standard deviation vector instead of covariance. You can compute the two via the following MATLAB command:
```matlab
[cormat, stdvec] = corrcov(covmat)
```
I have a vague memory about an old version of the library having this problem that you have raised, which we fixed a long time ago. We may not have released the latest fix to the release page (although the fix is likely in the repository's main branch). Given that the MATLAB version will be out soon, mid-next week, I do not see much benefit to releasing the patch separately. It will likely take the same amount of time as releasing the much better new version.
